linkerd/proxy/http/src/server.rs

```diff
@@ -145,120 +124,72 @@ where
     }
 
     fn call(&mut self, io: PrefixedIo<I>) -> Self::Future {
-        let version = HttpVersion::from_prefix(io.prefix());
-        match version {
-            Some(HttpVersion::Http1) => {
-                debug!("Handling as HTTP");
-                let http1 = if let Some(svc) = self.http1.clone() {
-                    trace!("HTTP service already exists");
-                    svc
-                } else {
-                    trace!("Building new HTTP service");
-                    let svc = self
-                        .new_http
-                        .new_service((HttpVersion::Http1, self.target.clone()));
-                    self.http1 = Some(svc.clone());
-                    svc
-                };
```

Comment:
nice to see that this logic is no longer necessary --- this always felt a little messy to me